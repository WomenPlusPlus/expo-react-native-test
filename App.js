import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import ExampleComponent from './ExampleComponent';

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Here is an example React Native App made via Expo with tests using Jest</Text>
      <ExampleComponent arg="3"></ExampleComponent>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
