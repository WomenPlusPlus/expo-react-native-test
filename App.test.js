import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

// Unit Test
describe('<App />', () => {
  it('has 2 children', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(2);
  });
});

// Snapshot Test
test('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
