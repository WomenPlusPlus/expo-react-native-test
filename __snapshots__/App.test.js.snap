// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Here is an example React Native App made via Expo with tests using Jest
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Child component
    </Text>
    <Text>
      3
    </Text>
  </View>
</View>
`;
